datalake/src/main/java/com/sequenceiq/datalake/service/sdx/flowwait/task/SdxWaitTaskForCloudbreakFlow.java

```diff
@@ -0,0 +1,59 @@
+package com.sequenceiq.datalake.service.sdx.flowwait.task;
+
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+import com.dyngr.core.AttemptResult;
+import com.dyngr.core.AttemptResults;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.service.sdx.PollingConfig;
+import com.sequenceiq.datalake.service.sdx.flowcheck.CloudbreakFlowService;
+import com.sequenceiq.datalake.service.sdx.flowcheck.FlowState;
+
+/**
+ * This wait task will wait for a cloudbreak flow or flow chain to finish, then checks the result of the flow.
+ */
+@Component
+@Scope("prototype")
```

Comment:
I would say it would be better if this would be a pojo and expect all dependency in the constructor, which would be provided by `SdxWaitTaskFactory`